src/main/scala/vectorpipe/ProcessOSM.scala

```diff
@@ -232,25 +218,7 @@ object ProcessOSM {
     import input.sparkSession.implicits._
     val st_pointToGeom = org.apache.spark.sql.functions.udf { pt: jts.Point => pt.asInstanceOf[jts.Geometry] }
 
-    // The following type conversion needed in case input comes from Change source
-    val elements =
-      Try(input.schema
-               .find(_.name == "members")
-               .get
-               .dataType
-               .asInstanceOf[ArrayType]
-               .elementType
-               .asInstanceOf[StructType]
-               .find(_.name == "type")
-               .get) match {
-        case Failure(_) => throw new IllegalArgumentException(s"Could not get type of 'members' in schema ${input.schema}")
-        case Success(field) =>
-          if (field.dataType == ByteType)
-            input.withColumn("members", convertMembers('members))
-          else
-            input
-      }
-
+    val elements = elaborateMemberTypes(input)
```

Comment:
This only needs to be done for the elements passed to `reconstructRelationGeometries` (and it will immediately be undone by `preprocessRelations` using `compressMemberTypes`). Can we flip this around to convert to Bytes if necessary here?

Reply:
So should the conversion happen only in `reconstructRelationGeometries`?  There's something here that seems unnecessarily convoluted.  I'm OK to ensure that we adhere to the compressed version internally, but it seems like if we get an ORC-originated frame, we then compress the member type field, and then the frame goes over to `reconstructRelationGeometries` which promptly undoes that change?  Should we not just assume a compressed internal representation in all places within the library?  Is that what you've been asking to do all along?  Something here isn't entirely making sense to me...

Reply:
> and then the frame goes over to reconstructRelationGeometries which promptly undoes that change?

`reconstructRelationGeometries` assumes compressed member types (and currently applies the compression by calling `preprocessRelations`).

> Should we not just assume a compressed internal representation in all places within the library?

We should. This involves compressing member types originating from ORC files.